### ISSUE TITLE
Add option to create tar.gz files.

### ISSUE THREAD

**Issue by yesimon** [zestsoftware/zest.releaser#76]
I see in this line that maybe python 2.4 has some issues with creating tar.gz files https://github.com/zestsoftware/zest.releaser/blob/08bde24a6de34d144731662e69cbf7e599be4021/zest/releaser/release.py#L105

However since that's a really old python by now, can we add an option to upload tar files to pypi?

**Comment by reinout** [comment]
The current option forces setuptools to generate a zipfile. I guess that's not really needed anymore.

So... instead of adding a tar option, I'd rather say "we don't need to pass the zip option anymore". Setuptools will then use some default, which is probably tgz files.

@mauritsvanrees: tgz is probably fine now?

**Comment by mauritsvanrees** [comment]
Probably okay. I started writing that we should still check for the Python version: if it is 2.4.x we should still use zip.  But looking at our setup.py we only support Python 2.6 and 2.7.  So should be okay to remove our zip default.

There is a nagging doubt in the back of my mind, because I had some problems recently with a tarball that I created on my Mac and that could not be extracted on some Linux server.  Problem was that I was using the Mac standard `/usr/bin/tar` which is `bsdtar`. Switching to a port install of GNU tar fixed it.  But this was with a `git archive` command, not with `zest.releaser` or `setuptools`.  `setuptools` may be using some internal or base Python tar version. Not sure. But as two data points on PyPI: for setuptools both `.zip` and `.tgz` are uploaded, and for `pip` only `.tgz` is uploaded. I guess we would know if there were problems with `.tgz` and recent Python versions.

Long story short: should be okay. :-) 

BTW, I think whether zip or targz or something else is used by default depends on the OS.  Indeed, see https://docs.python.org/2/distutils/sourcedist.html. If you want to override the default behavior you can add a `setup.cfg` to your Python project and put this in it, with one or more formats:

```
[sdist]
formats = zip,gztar
```

**Comment by reinout** [comment]
Hm... I am getting all sorts of hard-to-debug doctest failures. The common problem.
Note: I made the change on master...

**Comment by yesimon** [comment]
Is this related to tgz files?

**Comment by mauritsvanrees** [comment]
Yep, the expected sometimes-I-fail-sometimes-I-dont related to 'tags dir does not exist' where we try to answer 'no, do not create it' first, and it creates it anyway. Something like that.

**Comment by reinout** [comment]
Ok, tests are OK again.

Released as 3.56 and.... it is a tgz automatically :-) 
https://pypi.python.org/packages/source/z/zest.releaser/zest.releaser-3.56.tar.gz

**Comment by mauritsvanrees** [comment]
@yesimon: the 'Yep' was for Reinout. As answer to you: No, the doctest failures are not related to tgz files, they have been here for a while, sometimes failing sometimes not.... :-(

See https://travis-ci.org/zestsoftware/zest.releaser/builds/54873008
